integrated_channels/integrated_channel/course_metadata.py

```diff
@@ -0,0 +1,127 @@
+"""
+Assist integrated channels with retrieving course metadata.
+
+Module contains resources for integrated pipelines to retrieve all the
+metadata for courses in the course catalog belonging to a particular
+enterprise customer.
+"""
+from __future__ import absolute_import, unicode_literals
+
+import json
+from logging import getLogger
+
+from django.contrib.auth.models import User
+from slumber.exceptions import SlumberHttpBaseException
+
+from enterprise.course_catalog_api import CourseCatalogApiClient
+from enterprise.models import EnterpriseCustomer
+
+
+EXCLUDED_COURSE_DETAIL_KEYS = [
+    'course_runs',
+]
+
+
+LOGGER = getLogger(__name__)
+
+
+def updated_course_run(course_details, course_run_details):
+    """
+    Update a course run-specific details dict with details from the base course.
+
+    Args:
+        course_details (dict): The details of the base course
+        course_run_details (dict): The details of the specific course run
+    """
+    course_details = course_details.copy()
+    course_run_details = course_run_details.copy()
+    for key, base_course_value in course_details.items():
+        if course_run_details.get(key) is None and key not in EXCLUDED_COURSE_DETAIL_KEYS:
+            course_run_details[key] = base_course_value
+    return course_run_details
+
+
+def get_course_list(user, enterprise_customer):
+    """
+    Determine which catalog belongs to a given Enterprise Customer and list the course runs it contains.
+
+    Args:
+        user: A Django user requesting the course list
+        enterprise_customer: The given Enterprise Customer
+
+    Returns:
+        iterable: An iterable containing the details of each course run.
+    """
+    catalog_id = enterprise_customer.catalog
+
+    if catalog_id is None:
+        LOGGER.info('No catalog attached to %s', enterprise_customer)
+    else:
+        client = CourseCatalogApiClient(user)
+
+        catalog_courses = client.get_single_catalog_courses(catalog_id)
+        LOGGER.info('Retrieving course list for catalog %s', catalog_id)
+
+        for course in catalog_courses:
+            course_key = course.get('key')
+            course_details = client.get_single_course_details(course_key)
+            for run in course_details.get('course_runs', []):
+                yield updated_course_run(course_details, run)
+
+
+class BaseCourseExporter(object):
+    """
+    Base class for course metadata exporters to implement a "send" method on.
+    """
+
+    data_transform = {}
+
+    def __init__(self, user, plugin_configuration):
+        """
+        Save the appropriate details for use elsewhere in the object.
+        """
+        self.user = user
+        self.enterprise_customer = plugin_configuration.enterprise_customer
+        self.plugin_configuration = plugin_configuration
+
+    def get_parsed(self):
```

Comment:
`get_parsed"...course_runs?  What are we actually doing when we say we are "parsing"?